lib/ui/dart_runtime_hooks.cc

```diff
@@ -203,17 +204,14 @@ void Logger_PrintString(Dart_NativeArguments args) {
 void SaveCompilationTrace(Dart_NativeArguments args) {
   uint8_t* buffer = nullptr;
   intptr_t length = 0;
+
   Dart_Handle result = Dart_SaveCompilationTrace(&buffer, &length);
-  if (Dart_IsError(result)) {
-    Dart_SetReturnValue(args, result);
-    return;
-  }
+  CHECK_AND_RETURN_DART_ENGINE_ERROR(args, result,
+                                     "Could not save compilation trace.");
```

Comment:
This solution feels somewhat heavy to me. Is this additional machinery really worth it? Is it possible to rewrite such code like this, to get same effect but more explicit and without introducing new abstractions:

Dart_Handle result = Dart_SaveCompilationTrace(&buffer, &length);
if (Dart_IsError(result)) {
  Dart_SetReturnValue(args, Dart_NewUnhandledExceptionError(result));
  return;
}

I guess you can also swap "result" for Dart_NewStringFromCString(), but wouldn't that lose the original error message returned by VM which we would like to keep?

Dart_Handle result = Dart_SaveCompilationTrace(&buffer, &length);
if (Dart_IsError(result)) {
  auto error = Dart_NewStringFromCString("Could not save compilation trace.");
  Dart_SetReturnValue(args, Dart_NewUnhandledExceptionError(error));
  return;
}

If you want to be really elaborate, you should probably join the two messages together in some way, to make it look something like this: "Could not save compilation trace: <original VM error message>".